#1867 Fixed left nav subtabs CSS in tearsheet

nmgokhale opened the pull request:
Fixes #1867 

![Screenshot 2024-04-12 at 2 10 48 PM](https://github.com/elyra-ai/canvas/assets/25124000/b4755184-843d-4777-8f8c-ae017c423bfb)

 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

matthoward366 reviewed (commented):
I wonder instead of having the bottom border if we shouldn't change the hover background color.  The rest looks good.

nmgokhale commented:
For non-selected buttons, removed bottom border on hover and added background color -

https://github.com/elyra-ai/canvas/assets/25124000/3713da46-ea02-4b9f-889e-dde6c4124d7a

matthoward366 commented on canvas_modules/common-canvas/src/common-properties/panels/subtabs/subtabs.scss:
```diff
@@ -83,4 +82,8 @@ $selected-tab-width: 3px;
 		padding: $spacing-05; // Override padding set by subtab
 	}
+
+	div.cds--tabs:not(.cds--tabs--contained) .cds--tabs__nav-item:not(.cds--tabs__nav-item--selected):not(.cds--tabs__nav-item--disabled):hover {
```
I don't usually like doing this might this might be a case where it's best to us !important.  So instead of this you can have
```
		button.properties-leftnav-subtab-item {
			&:hover {
				background-color: $layer-hover-01;
				border-block-end: none !important;
			}
```
Just make sure to add a comment to why we have it.

nmgokhale commented:
My local branch is out of sync with remote branch. I can't push anymore changes to this PR. I will close this PR and create a new one with same changes.
